src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php

```diff
@@ -304,24 +309,18 @@ protected function getCategoryListRepresentation(Request $request, $parentKey =
 
         $listBuilder->addSelectField($fieldDescriptors['depth']);
         $listBuilder->addSelectField($fieldDescriptors['parent']);
-        $listBuilder->addSelectField($fieldDescriptors['hasChildren']);
         $listBuilder->addSelectField($fieldDescriptors['locale']);
         $listBuilder->addSelectField($fieldDescriptors['defaultLocale']);
-
-        $listBuilder->addGroupBy($fieldDescriptors['id']);
+        $listBuilder->addSelectField($fieldDescriptors['lft']);
+        $listBuilder->addSelectField($fieldDescriptors['rgt']);
 
         if ($parentKey !== null) {
             $this->addParentSelector($parentKey, $listBuilder);
         }
 
-        // FIXME: don't do this.
-        $listBuilder->limit(100000);
```

Comment:
I've got the feeling that you can't delete that line... Now only 10 elements are returned, and the children are not loaded. So if you have 10 root categories none of the children are loaded. Maybe it works when you set the limit to `null`? The no limit at all should be applied, and it is not a magic number like now. The categories should be improved in that way quite soon anyway.